### ISSUE TITLE
Make ha_tracker_failover_timeout configurable per-tenant

### ISSUE THREAD

**Issue by bboreham** [grafana/mimir#4066]
#### Is your feature request related to a problem? Please describe.

The HA-tracker inside distributor will look to change tenant if it hasn't received a message inside `ha_tracker_failover_timeout`, which defaults to 30s.
Suppose a tenant is set up with one target, scraping at 1 minute intervals, then this default does not suit: we should be able to configure the timeout per-tenant. 

#### Describe the solution you'd like

Add `ha_tracker_failover_timeout` to runtime config.  Note the existing ones are a bit shorter, e.g. `ha_cluster_label`.

If one tenant has multiple clusters and only a few of them send very slowly, then we might need more fine-grained configuration.

#### Describe alternatives you've considered

Have it auto-tune - see comment from @pracucci below.

**Comment by pracucci** [comment]
Would be very nice if we wouldn't have to change it manually in the config, but we could auto-detect the tenant scrape interval (or highest value if customer has multiple scraping jobs with different intervals), so that we can apply the failover timeout equal to `max(configured timeout, highest scrape interval * delta)`.

**Comment by bboreham** [comment]
Yes I agree.  We can't assume data is coming from a "scrape", but we could track the recent rate of arrival of messages at each distributor.  

Would we keep that in the KV-store?

**Comment by pracucci** [comment]
I'm also wondering if we could improve the remote write protocol to pass some context information, like the min/max scrape interval.

**Comment by bboreham** [comment]
Typical clients are scraping multiple targets, so the scrape interval on its own is not useful.

**Comment by duricanikolic** [comment]
I think this issue is related to the one opened by @mattmendick a couple of years before: https://github.com/cortexproject/cortex/issues/3812

**Comment by tcard** [comment]
> Would be very nice if we wouldn't have to change it manually in the config, but we could auto-detect the tenant scrape interval (or highest value if customer has multiple scraping jobs with different intervals), so that we can apply the failover timeout equal to `max(configured timeout, highest scrape interval * delta)`.

I've extracted a new issue for this:

* https://github.com/grafana/mimir/issues/12108
